Objects/weakrefobject.c

```diff
@@ -874,6 +875,31 @@ PyWeakref_GetObject(PyObject *ref)
 static void
 handle_callback(PyWeakReference *ref, PyObject *callback)
 {
+    /* If the garbage collector support is not properly implemented on
+     * some classes that are involved in a reference cycle, a weak
+     * reference may try to invoke a callback object that is being
+     * cleaned (tp_clear) by the garbage collector and it may be in an
+     * inconsistent state. As the garbage collector explicitly does
+     * not invoke callbacks that are part of the same cycle isolate as
```

Comment:
"cycle isolate" doesn't appear as a common term elsewhere in the source (if it is meant to be a term and not a typo). Perhaps rephrase it as "does not invoke callbacks when they are being destroyed at the same time as ..."? Or explain the term somewhere.